src/main/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheck.java

```diff
@@ -125,42 +106,49 @@
 
     @Override
     public void beginTree(DetailAST rootAST) {
-        publicTypeFound = false;
-        lineNumberTypeMap.clear();
-
         DetailAST currentNode = rootAST;
+        boolean publicTypeFound = false;
+        DetailAST firstType = null;
```

Comment:
can it be changed to something like
```
DetailAST firstType = TokenUtil.findFirstTokenByPredicate(rootAST, this::isTypeDef)
```
? 
It will look better as current logic

Reply:
We are iterating through siblings of `rootAST` as we are dealing with top-level declarations. `findFirstTokenByPredicate` will iterate through children of `rootAST` which is not what we want.

Trying `TokenUtil.findFirstTokenByPredicate(rootAST.getParent(), this::isTypeDef)` leads to some null pointer exceptions, presumably due to how `rootAST`'s parent is being handled.